src/include/sof/schedule/twb_schedule.h

```diff
@@ -0,0 +1,81 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/*
+ * Copyright(c) 2023 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@intel.com>
+ */
+
+#ifndef __SOF_SCHEDULE_TWB_SCHEDULE_H__
+#define __SOF_SCHEDULE_TWB_SCHEDULE_H__
+
+#include <rtos/task.h>
+#include <stdint.h>
+
+/**
+ * @brief Task With Budget (TWB) Scheduler
+ *
+ * TWB scheduler is a scheduler that creates a separate preemptible Zephyr thread for each SOF task that has a pre-allocated MCPS budget renewed with every system tick.
+ * The TWB scheduler assigns either MEDIUM_PRIORITY or LOW_PRIORITY to the task thread based on the budget left in the current system tick.
+ * It allows for opportunistic execution if there is no other ready task with a higher priority while the budget is already spent.
+ *
+ * Examples of tasks with budget include Ipc Task and Idc Task.
+ *
+ * The TWB scheduler has two key parameters assigned:
+ * - cycles granted: the budget per system tick
+ * - cycles consumed: the number of cycles consumed in a given system tick for task execution
+ *
+ * The number of cycles consumed is reset to 0 at the beginning of each system tick, renewing the TWB budget.
+ * When the number of cycles consumed exceeds the cycles granted, the task is switched from MEDIUM to LOW priority.
+ * When the task with budget thread is created, the MPP Scheduling is responsible for setting the thread time slice equal to the task budget, along with setting a callback on time slice timeout.
+ * Thread time slicing guarantees that the Zephyr scheduler will interrupt execution when the budget is spent, so the MPP Scheduling timeout callback can re-evaluate the task priority.
+ *
+ * If there is a budget left in some system tick (i.e., the task spent less time or started executing close to the system tick that preempts execution), it is reset and not carried over to the next tick.
+ *
+ * Mode info: https://thesofproject.github.io/latest/architectures/firmware/sof-zephyr/mpp_layer/mpp_scheduling.html
+ */
+
+/**
+ * \brief default static stack size for each TWB thread
+ */
+#define ZEPHYR_TWB_STACK_SIZE 8192
+
+/**
+ * \brief default priority of the TWB threads in the system.
+ */
+#define ZEPHYR_TWB_THREAD_PRIORITY (CONFIG_NUM_PREEMPT_PRIORITIES - 3)
+#define ZEPHYR_TWB_THREAD_PRIORITY_LOW (CONFIG_NUM_PREEMPT_PRIORITIES - 2)
```

Comment:
Priority of threads are spread between many files, it's hard to maintenance
We need a single place for all priorities - LL, DP, TWB and P4Q and similar

Reply:
There is no specific requirement other than:

Low Latency audio data processing tasks (LL) - **high** priority,
Tasks with Budget (TwB) - **medium** priority,
audio Data Processing tasks (DP) - **low** priority,
background (idle) priority tasks
https://thesofproject.github.io/latest/architectures/firmware/sof-zephyr/mpp_layer/mpp_scheduling.html

So I just picked "medium" value between LL and DP, we can discuss specific values here.

Reply:
added kconfig.threads_prio with priorities used in SOF